### PR TITLE
EVAKA-4045: Adjust service voucher decision texts

### PR DESCRIPTION
**Pull request by kustaahuhtala**
#### Summary

- Widen create new person modal
- Reorganise PDF creation and testing, add voucher generation test
- Do not use DaycareFormV0 when generating PDFs
- Refactor and simplify DecisionService usage
- Update private service voucher decision texts

#### How to test

- Run tests in `PDFServiceTest`. Tests will print paths to generated PDFs in the logs. Check both, Finnish and Swedish versions
- Uncomment `//File("/tmp/fee_decision_test.pdf").writeBytes(pdfBytes)` in `PdfServiceTest` and run `generateFeeDecisionPdf smoke test`. (Yes, I know, naming sucks). Check the generated fee decision in `/tmp/fee_decision_test.pdf`. Change the language in the test to generate Swedish version of fee decision.

### CODE REVIEW

File: service/src/main/kotlin/fi/espoo/evaka/decision/DecisionService.kt [hunk 1]
```diff
@@ -65,136 +64,184 @@ class DecisionService(
...
+        application.otherGuardianId != null &&
+        !personService.personsLiveInTheSameAddress(
+            tx,
+            application.guardianId,
+            application.otherGuardianId
         )
 
-        val pages = templates.mapIndexed { i, template ->
-            Page(
+    companion object {
```
Comment (jaahon) [inline service/src/main/kotlin/fi/espoo/evaka/decision/DecisionService.kt:139]: Could these be just top level functions or is there some meaning behind putting these inside `companion object`?

**Review by jaahon — COMMENTED** [review]
Good job! I have a couple of minor comments below. 

Also, it seems like you still have some stuff commented out in the code and templates. You might want to clean them up.

**Comment by jaahon** [comment]
In all honestly I kind of just skimmed through the templates, but the final PDFs looked good (didn't measure millimeters though 😄 )

File: service/src/main/kotlin/fi/espoo/evaka/decision/DecisionService.kt [hunk 2]
```diff
@@ -65,136 +64,184 @@ class DecisionService(
...
-            decision.type != DecisionType.CLUB &&
-            application.otherGuardianId != null &&
-            !personService.personsLiveInTheSameAddress(
-                tx,
-                application.guardianId,
-                application.otherGuardianId
+        val unitManager = (
+            tx.handle.getUnitManager(decision.unit.id)
+                ?: throw NotFound("Daycare manager not found with daycare id: ${decision.unit.id}.")
+            )
```
[the remark this answers is not in the document]
Comment (kustaahuhtala) [reply service/src/main/kotlin/fi/espoo/evaka/decision/DecisionService.kt]: Will fix this.

File: service/src/main/kotlin/fi/espoo/evaka/decision/DecisionService.kt [hunk 3]
```diff
@@ -65,136 +64,184 @@ class DecisionService(
...
                 ?: throw NotFound("Other guardian not found with id: ${application.otherGuardianId}")
 
-            createDecisionPdf(tx, decision, application, otherGuardian, child)
-                .let { uri -> updateDecisionOtherGuardianDocumentUri(tx.handle, decisionId, uri) }
+            updateDecisionOtherGuardianDocumentUri(
+                tx.handle,
+                decisionId,
+                createAndUploadDecision(
+                    decision, application, otherGuardian, child, decisionLanguage, guardian, unitManager
+                )
```
[the remark this answers is not in the document]
Comment (kustaahuhtala) [reply service/src/main/kotlin/fi/espoo/evaka/decision/DecisionService.kt]: Will be changed.

File: service/src/main/kotlin/fi/espoo/evaka/decision/DecisionService.kt (hunk 1 above)
Comment (kustaahuhtala) [reply service/src/main/kotlin/fi/espoo/evaka/decision/DecisionService.kt]: Yes they could actually, I don't know what was I thinking here.

**Review by jaahon — APPROVED** [review]
🎖️

**Pull request merged**